pkg/handlers/validate.go

```diff
@@ -15,12 +16,35 @@ import (
 // 	k8s.io/kubernetes/pkg/util/validation/validation.go
 var validDNS = regexp.MustCompile(`^[a-z0-9]([-a-z0-9]*[a-z0-9])?$`)
 
-// ValidateDeployRequest validates that the service name is valid for Kubernetes
-func ValidateDeployRequest(request *types.FunctionDeployment) error {
-	matched := validDNS.MatchString(request.Service)
+// validates that the service name is valid for Kubernetes
+func validateService(service string) error {
+	matched := validDNS.MatchString(service)
 	if matched {
 		return nil
 	}
 
-	return fmt.Errorf("(%s) must be a valid DNS entry for service name", request.Service)
+	return fmt.Errorf("service: (%s) is invalid, must be a valid DNS entry", service)
+}
+
+// ValidateDeployRequest validates that the service name is valid for Kubernetes
+func ValidateDeployRequest(request *types.FunctionDeployment) error {
```

Comment:
I can guess why you did this, however as a rule, we want new contributions to match previous ones.

So generally, a validation rule will return the first error that is matched rather than N errors, it also makes it easier for callers to validate and handle predictable, stable error messages.

I'd prefer to see this code return: `validation failed: service: is required`, followed by a separate error if the user adds the service, and has no value for "image".

Instead of `validation failed: service: is required; image: is required
`